components/SingleArticlePage.jsx

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from "react"
+import { useParams } from "react-router-dom"
+import { fetchArticleById } from "../utils/api"
+
+export default function SingleArticlePage(){
+    const [singleArticle, setSingleArticle] = useState({})
+    const [isLoading, setIsLoading] = useState(false)
+    const [isError, setIsError] = useState(false)
+    const { article_id} = useParams()
+
+    useEffect(() => {
+        if (article_id){
```

Comment:
You should always have an article id because without it the router will never match this route